File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def helper(self, node: Optional[TreeNode], currlvl: int) -> int:
        if node == None:
            return currlvl
        currlvl += 1
        l = self.helper(node.left, currlvl) 
        r = self.helper(node.right, currlvl)       
        return max(l, r)
    
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.helper(root, 0)







    

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(std::vector<int> &nums) 
    {
        int left = 0;
        int right = nums.size() - 1;
        int min_val = nums[left];
        int middle = (left + right) / 2;

        // if (right == 0)
        //     return nums[0];
        while (left <= right)
        {
            if (min_val <= nums[middle])
            {
                left = middle + 1;
            }
            else if (min_val > nums[middle])
            {
                right = middle - 1;
                min_val = nums[middle];
            }
            middle = (left + right) / 2;
        }
        return min_val;
    }
};
